python/ray/dag/compiled_dag_node.py

```diff
@@ -1175,13 +1249,33 @@ def execute(
 
         self._get_or_compile()
 
+        self._check_inputs(args, kwargs)
```

Comment:
Put this to the end of `_get_or_compile()` since it is always called?

Reply:
Prefer to do it separately because `get_or_compile` shouldn't need to depend on the inputs to `dag.execute()` (which can change each time dag.execute() is called).